xwiki-platform-core/xwiki-platform-mentions/xwiki-platform-mentions-notifications/src/test/java/org/xwiki/mentions/templates/MentionTest.java

```diff
@@ -0,0 +1,150 @@
+        event1.setDate(eventDate);
+        event1.setUser(userPage1);
+
+        DefaultEvent event2 = new DefaultEvent();
+        event2.setDate(eventDate);
+        event2.setUser(userPage2);
+
+        // Create and save page 1 with a title.
+        XWikiDocument xWikiDocument1 = this.xwiki.getDocument(page1, this.context);
+        xWikiDocument1.setTitle("Page 1 Title");
+        this.xwiki.saveDocument(xWikiDocument1, this.context);
+        // Create and save page 2 without a title.
+        XWikiDocument xWikiDocument2 = this.xwiki.getDocument(page2, this.context);
+        this.xwiki.saveDocument(xWikiDocument2, this.context);
+
+        CompositeEvent value = new CompositeEvent(event1);
+        value.getEvents().add(event2);
+
+        Map<Object, Object> compositeEventParams = new HashMap<>();
+        MentionView mentionView1 = new MentionView()
+            .setAuthorURL("/U1")
+            .setDocumentURL("/page1")
+            .setDocument(xWikiDocument1);
+        compositeEventParams.put(event1, mentionView1);
+        MentionView mentionView2 = new MentionView()
+            .setAuthorURL("/U2")
+            .setDocument(xWikiDocument2)
+            .setDocumentURL("/page2");
+        compositeEventParams.put(event2, mentionView2);
+
+        velocityContext.put("compositeEvent", value);
+        velocityContext.put("compositeEventParams", compositeEventParams);
+
+        registerVelocityTool("escapetool", new EscapeTool());
+        registerVelocityTool("stringtool", new StringUtils());
+
+        this.componentManager.registerComponent(ScriptService.class, "icon", this.iconManager);
+        this.componentManager.registerComponent(ScriptService.class, "date", this.dateScriptService);
+        this.componentManager.registerComponent(ScriptService.class, "localization", this.localizationScriptService);
+
+        when(this.iconManager.renderHTML("bell")).thenReturn("bell-icon");
+        when(this.iconManager.renderHTML("page")).thenReturn("page-icon");
+        when(this.dateScriptService.displayTimeAgo(eventDate)).thenReturn("one hundred years ago");
+        when(this.localizationScriptService.render("mentions.event.mention.summary.plural", singletonList(2)))
+            .thenReturn("You have received 2 mentions.");
+        when(this.localizationScriptService.render("mentions.event.mention.description"))
+            .thenReturn("mentioned you on page");
+        when(this.localizationScriptService.render("mentions.event.mention.type")).thenReturn("Mentions");
+
+        String actual = cleanup(templateManager.render("templates/mentions/mention.vm"));
+
+        String expected =
+            cleanup(IOUtils.toString(getClass().getResourceAsStream("/templates/mentions/mention.mention.html"),
+                UTF_8));
+
+        assertEquals(expected, actual);
+    }
+
+    private String cleanup(String render)
+    {
+        return render.trim().replaceAll("\\s+", " ");
+    }
```

Comment:
This should be moved to a test util method, we need to define where.

Reply:
This is already a code duplication and it will be useful for future template page tests.

Reply:
see https://matrix.to/#/!ikPtGZaGWtyblizzlR:matrix.xwiki.com/$16195289384221TGmYu:matrix.xwiki.com?via=matrix.org&via=matrix.xwiki.com&via=dodsorf.as